src/daft-io/src/object_store_glob.rs

```diff
@@ -404,6 +404,29 @@ pub async fn glob(
     };
     let glob = glob.as_str();
 
+    // We need to do some validation on the glob pattern before compiling it, since the globset crate is very permissive
+    // and will happily compile patterns that don't make sense without throwing an error.
+    fn verify_glob(glob: &str) -> super::Result<()> {
```

Comment:
Nit: we can move this out of the function, and just run Rust unit-tests!

Reply:
Ah, makes a lot of sense.

Wasn't familiar with how Rust unit tests worked, have moved the testing logic for the `verify_glob` function to `object_store_glob.rs`.